tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -68,6 +68,8 @@ public ImmutableTracerSettings(TracerSettings settings)
             // we cached the static instance here, because is being used in the hotpath
             // by IsIntegrationEnabled method (called from all integrations)
             _domainMetadata = DomainMetadata.Instance;
+
+            ExpandRouteParametersEnabled = settings.ExpandRouteParametersEnabled || !RouteTemplateResourceNamesEnabled;
```

Comment:
In reality, the code paths don't intersect, and this will all clean up more cleanly once `RouteTemplateResourceNamesEnabled` is removed, but logically if `RouteTemplateResourceNamesEnabled` is `false`, then `ExpandRouteParametersEnabled` should be _true_ (as that was the "old" behaviour, sort of)